### PR TITLE
Docs for the core folder, and gulp task for jsdoc and docstrap

### PR DESCRIPTION
**Pull request by alvinsight**

### CODE REVIEW

File: gulp/tasks/jsdoc.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+gulp.task('jsdocs',function(){
+
+	// here we use a globbing pattern to match everything inside the 'documentation' folder
+	var folders = [
+    './documentation/**'
+    ];
+
+
+    var docStrapTemplate = {
```
Comment (englercj) [inline gulp/tasks/jsdoc.js]: Isn't this a duplicate of the conf file in `gulp/util/jsdoc.conf.json`?

https://github.com/GoodBoyDigital/pixi.js/blob/dev/gulp/util/jsdoc.conf.json

Doc generation right now is through `npm run docs` which works fine. Do we need a gulp task as well?

File: gulpfile.js [hunk 2]
```diff
@@ -30,3 +30,6 @@ requireDir('./gulp/tasks', { recurse: true });
 
 // default task
 gulp.task('default', ['jshint', 'build']);
+
+// you can directly use 'gulp jsdocs' to generate the docs
+gulp.task('docs','jsdocs');
```
Comment (englercj) [inline gulpfile.js]: this would have to be an array of task dependencies: `gulp.task('docs', ['jsdocs']);`

Comment (alvinsight) [inline gulpfile.js]: Just fixed it, thanks :)

**Comment by englercj** [comment]
I've updated the jsdoc config to run on the entire source folder (using `npm run docs`) as well as fixed the jsdoc errors.

**Pull request merged**